src/main/resources/default/assets/common/core.js

```diff
@@ -71,3 +71,92 @@ sirius.postJSON = function(url, params) {
         return response.json();
     });
 }
+
+/**
+ * Throttles a javascript function by calling it only every x milliseconds.
+ * <p>
+ * The throttled function is called once immediately, and then every x milliseconds.
+ *
+ * @param fn        the function to be throttled
+ * @param threshold the time delaying each function call (in milliseconds).
+ */
+sirius.throttle = function (fn, threshold) {
+    let last;
+    let deferTimer;
+
+    return function () {
+        const context = this;
+        const now = Date.now();
+        const args = arguments;
+
+        if (last && now < last + threshold) {
+            // hold on to it
+            clearTimeout(deferTimer);
+
+            deferTimer = setTimeout(function () {
+                last = now;
+                fn.apply(context, args);
+            }, threshold);
```

Comment:
shouldnt this timeout be the remaining time instead of the treshhold? else it behaves more like debounce -

e.g. if i call it a 0, with a treshhold of 1000, and again at 800, the timeout is set to execute at 1800, and the correct behavior would be 1000, i think?

Reply:
You're right, i corrected that